commands/deploy_command.go

```diff
@@ -221,24 +225,95 @@ func (c *DeployCommand) Execute(args []string) ExecutionStatus {
 	mtaElementsCalculator := mtaElementsToAddCalculator{shouldAddAllModules: false, shouldAddAllResources: false}
 	mtaElementsCalculator.calculateElementsToDeploy(optionValues)
 
-	mtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
+	rawMtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
 	if err != nil {
 		ui.Failed("Error retrieving MTA: %s", err.Error())
 		return Failure
 	}
 
+	isUrl := false
+	mtaArchive := ""
+	switch castedMtaArchive := rawMtaArchive.(type) {
+	case *url.URL:
+		isUrl = true
+		mtaArchive = castedMtaArchive.String()
+	case string:
+		mtaArchive = castedMtaArchive
+	default:
+		return Failure
```

Comment:
Should we add a message before returning Failure: `ui.Failed(...)`?

Reply:
This case shouldn't happen at all. I will remove it